### PR TITLE
Selectively enable wrapping constructors

### PR DESCRIPTION
**Pull request by ddn0**
Libraries like outcome depend on implicit conversions. Forwarding
constructors like

  template<typename... Args>
  T(Args&&...): member(std::forward<Args>(args)...)

can shadow desired implicit conversions, even if
member(std::forward<Args>(args)...) itself is not valid.

Instead, help the C++ overload selection process by pruning non-viable
template instantiations with some concept checking.

### CODE REVIEW

**Review by roshandathathri — APPROVED** [review]
LGTM. Thanks!

File: conda_recipe/meta.yaml [hunk 1]
```diff
@@ -80,12 +81,13 @@ outputs:
...
         - libxml2 {{ libxml2 }}
         - llvmdev {{ llvmdev }}
-        - ncurses {{ ncurses }}
         - nlohmann_json {{ nlohmann_json }}
+        - numpy {{ numpy }}
         - openblas {{ openblas }}
         - zlib {{ zlib }}
         - python {{ python }}
       run:
+        - numpy
```
Comment (arthurp) [inline conda_recipe/meta.yaml]: I'm not sure if numpy is needed here. But it shouldn't be a problem.

File: scripts/build_in_container.py [hunk 2]
```diff
@@ -116,6 +117,7 @@ def make_docker_run_command(katana_version):
         )
         print("or modify this command to access the build environment in other ways.")
 
+    return 0
 
 if __name__ == "__main__":
-    main()
+    sys.exit(main())
```
Comment (arthurp) [inline scripts/build_in_container.py]: Sorry about this. :-/ I really messed this up.

**Review by arthurp — APPROVED** [review]
The numpy related changes look good. I do think there is one more thing that will help though:

```patch
iff --git a/conda_recipe/conda_build_config.yaml b/conda_recipe/conda_build_config.yaml                                           
index 001dc94ea..7489f4006 100644                                                                                                  
--- a/conda_recipe/conda_build_config.yaml                                                                                         
+++ b/conda_recipe/conda_build_config.yaml                                                                                         
@@ -4,7 +4,7 @@                                                                                                                    
 #  allowed version range.                                                                                                         
                                                                                                                                   
 numpy:                                                                                                                            
-  - 1.21                                                                                                                          
+  - 1.22                                                                                                                          
                                                                                                                                   
 python:                                                                                                                           
   - 3.8.* *_cpython                                                                                                               
@@ -33,7 +33,7 @@ pin_run_as_build:                                                                                                
   numba:                                                                                                                          
     max_pin: x                                                                                                                    
   numpy:                                                                                                                          
-    max_pin: x                                                                                                                    
+    max_pin: x.x                                                                                                                  
   pyarrow:                                                                                                                        
     max_pin: x                                                                                                                    
   bliss:                         
```

File: conda_recipe/conda_build_config.yaml [hunk 3]
```diff
@@ -31,9 +31,9 @@ pin_run_as_build:
   nlohmann_json:
     max_pin: x
   numba:
-    max_pin: x
+    max_pin: x.x
```
Comment (arthurp) [inline conda_recipe/conda_build_config.yaml]: I think this is reasonable. But I'm curious if there is a specific reason to increase this pinning.

**Comment by ddn0** [comment]
Carved off the CI fixing work to a separate PR: https://github.com/KatanaGraph/katana/pull/814

**Pull request merged**